k8s/controllers/solution/instance_queueing_controller.go

```diff
@@ -67,10 +69,25 @@ func (r *InstanceQueueingReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 	} else { // remove
 		reconciliationType = metrics.DeleteOperationType
-		operationName := fmt.Sprintf("%s/%s", constants.InstanceOperationNamePrefix, constants.ActivityOperation_Delete)
-		deploymentOperationType, reconcileResult, err = r.dr.AttemptUpdate(ctx, instance, true, log, instanceOperationStartTimeKey, operationName)
-		if err != nil {
-			resultType = metrics.ReconcileFailedResult
+		log.Info("Reconcile removing instance: " + req.Name + " in namespace " + req.Namespace)
+		// check the finalizer - uninstall finalizer if exists, set finalizer to nil
+		if utils.ContainsString(instance.GetFinalizers(), os.Getenv(constants.DeploymentFinalizer)) {
+			// set finalizer to nil
+			log.Info("Reconcile removing instance finalizier: " + req.Name + " in namespace " + req.Namespace)
+			patch := client.MergeFrom(instance.DeepCopy())
+			instance.SetFinalizers([]string{})
+			if err := r.Patch(ctx, instance, patch); err != nil {
+				log.Error(err, "Failed to patch instance finalizers")
+				resultType = metrics.ReconcileFailedResult
```

Comment:
If patch failed, do we need to set requeueAfter in the reconcileresult?

Reply:
return non-nil error is another kind of requeue.